### PR TITLE
feat: added example for document saver

### PR DESCRIPTION
**Pull request by vishwarajanand**
[b/348044051](http://b/348044051)

#### Changes 
 
  - Missing `await` in previous example for `AlloyDBLoader.create`
  - Hardcoded table name now uses `TABLE_NAME` var set by the user
  - Added an example for Document Saver

#### Test

Created a Table in Alloy DB like this:

```
CREATE TABLE vectorstore_custom (
    product_name TEXT NOT NULL,
    content TEXT NOT NULL,
    description TEXT NOT NULL,
    id SERIAL PRIMARY KEY,
    metadata JSON
)
```

### CODE REVIEW

File: docs/document_loader.ipynb [hunk 1]
```diff
@@ -321,15 +321,131 @@
...
     ")\n",
     "docs = await loader.aload()\n",
     "print(docs)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create AlloyDBDocumentSaver"
```
Comment (averikitsch) [inline docs/document_loader.ipynb:338]: nit: use H2 header and maybe rename "Basic usage" to "Create AlloyDBLoader"

File: docs/document_loader.ipynb [hunk 2]
```diff
@@ -321,15 +321,131 @@
...
+    "    metadata_json_column=\"metadata\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Save Documents to default table\n",
+    "The Alloy DB Document saver allows to save a list of Documents to the table using the first column as page_content and all other columns as metadata. \n",
```
Comment (averikitsch) [inline docs/document_loader.ipynb:364]: Move this description under the "### Create AlloyDBDocumentSaver" section

Comment (averikitsch) [inline docs/document_loader.ipynb:364]:
```suggestion
    "The `AlloyDBDocumentSaver` allows for saving of pre-processed documents to the table using the first column as page_content and all other columns as metadata. This table can easily be loaded via a Document Loader or updated to be a VectorStore.\n",
```

File: docs/document_loader.ipynb [hunk 3]
```diff
@@ -321,15 +321,131 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_google_alloydb_pg import AlloyDBDocumentSaver\n",
+    "\n",
+    "# Creating a basic AlloyDBDocumentSaver object\n",
```
Comment (averikitsch) [inline docs/document_loader.ipynb:349]: Can we also show how to create a table using method: `init_document_table()`?

File: docs/document_loader.ipynb [hunk 4]
```diff
@@ -321,15 +321,131 @@
...
+    "    ),\n",
+    "]\n",
+    "await saver.aadd_documents(test_docs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The added Document is returned with AlloyDBLoader"
```
Comment (averikitsch) [inline docs/document_loader.ipynb:399]:
```suggestion
    "### Load the documents with AlloyDBLoader"
```

File: docs/document_loader.ipynb [hunk 5]
```diff
@@ -321,15 +321,131 @@
...
+   "outputs": [],
+   "source": [
+    "await saver.adelete(test_docs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The added Documents are deleted, hence NOT returned with AlloyDBLoader"
```
Comment (averikitsch) [inline docs/document_loader.ipynb:435]: nit: I don't think we need this section

File: docs/document_loader.ipynb [hunk 6]
```diff
@@ -321,15 +321,131 @@
...
+    "### The added Document is returned with AlloyDBLoader"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "loader = await AlloyDBLoader.create(engine, table_name=TABLE_NAME)\n",
```
Comment (averikitsch) [inline docs/document_loader.ipynb:408]: Can we add a note that we are using the table_name with the loader over the query to load the whole table?

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: docs/document_loader.ipynb (hunk 3 above)
Comment (vishwarajanand) [reply docs/document_loader.ipynb:349]: Added it just after engine initialization

**Pull request merged**